common/main/java/com/couchbase/lite/internal/replicator/AbstractCBLWebSocket.java

```diff
@@ -175,7 +184,10 @@ public void onMessage(@NonNull WebSocket webSocket, @NonNull ByteString bytes) {
         public void onClosing(@NonNull WebSocket webSocket, int code, @NonNull String reason) {
             Log.d(TAG, "%s#OkHTTP closing: %s", AbstractCBLWebSocket.this, reason);
             synchronized (getPeerLock()) {
-                if (!state.setState(State.CLOSE_REQUESTED)) { return; }
+                if (!state.setState(State.CLOSE_REQUESTED))  {
+                    if (state.assertState(State.CLOSED)) { webSocket.cancel(); }
```

Comment:
I couldn't think about the case that the state is already CLOSED while OKHTTP is closing. However it doesn't hurt to put the logic here for precaution.